modules/highgui/src/window_QT.cpp

```diff
@@ -299,6 +299,13 @@ CV_IMPL void cvDisplayOverlay(const char* name, const char* text, int delayms)
         Q_ARG(QString, QString(name)),
         Q_ARG(QString, QString(text)),
         Q_ARG(int, delayms));
+
+    if (delayms > 0)
+    {
+        QTimer::singleShot(delayms, [name]() {
+            cvUpdateWindow(name);  // Force redraw
+        });
+    }
```

Comment:
What if just add "update" to the end of `DefaultViewPort::stopDisplayInfo()`? The method is involed by existing timer. No need to create a new one.